src/SFML/System/Clock.cpp

```diff
@@ -31,36 +31,103 @@
 
 namespace sf
 {
+
 ////////////////////////////////////////////////////////////
 Clock::Clock() :
-m_startTime(ClockImpl::now())
+m_refPoint(ClockImpl::now()),
+m_pausePoint()
 {
+
 }
 
 
 ////////////////////////////////////////////////////////////
 Time Clock::getElapsedTime() const
 {
-    return durationToTime(ClockImpl::now() - m_startTime);
+    if (isRunning())
+    {
+        return durationToTime(ClockImpl::now() - m_refPoint);
+    }
+    else
+    {
+        return durationToTime(m_pausePoint - m_refPoint);
+    }
+}
+
+
+////////////////////////////////////////////////////////////
+void Clock::setRunning(bool running)
+{
+    if (running)
+    {
+        start();
+    }
+    else
+    {
+        pause();
+    }
+}
+
+
+////////////////////////////////////////////////////////////
+void Clock::start()
+{
+    if (!isRunning())
+    {
+        m_refPoint += ClockImpl::now() - m_pausePoint;
+
+        m_pausePoint = ClockImpl::time_point();
+    }
+}
+
+
+////////////////////////////////////////////////////////////
+void Clock::pause()
+{
+    if (isRunning())
+    {
+        m_pausePoint = ClockImpl::now();
+    }
 }
 
 
 ////////////////////////////////////////////////////////////
 Time Clock::restart()
 {
-    const ClockImpl::time_point now = ClockImpl::now();
-    Time elapsed = durationToTime(now - m_startTime);
-    m_startTime = now;
+    const Time current = getElapsedTime();
+
+    m_pausePoint = ClockImpl::time_point();
+
+    m_refPoint = ClockImpl::now();
```

Comment:
Could you remove a few of these extra empty newlines to better match the style of existing code? Most SFML code puts expressions on consecutive lines instead of double-spacing everything.

Reply:
Seems reasonable, gonna address that.